spec/functions/fqdn_rand_string_spec.rb

```diff
@@ -77,9 +51,7 @@ def fqdn_rand_string(max, args = {})
     charset = args[:charset]
     extra = args[:extra_identifier] || []
 
-    scope = PuppetlabsSpec::PuppetInternals.scope
-    scope.stubs(:[]).with("::fqdn").returns(host)
-    scope.stubs(:lookupvar).with("::fqdn").returns(host)
+    scope.stubs(:lookupvar).with("::fqdn", {}).returns(host)
```

Comment:
When you look at the tests, you'll see that the tests compare the output from different hosts. This precludes using the `let(:facts)` thingy. I've added a comment and unstub it too.